### PR TITLE
JS DOM

### PR DESCRIPTION
**Pull request by ustymchyk**
### JS DOM

[Demo](https://ustymchyk.github.io/js-dom/) |
[Code base](https://github.com/ustymchyk/js-dom/tree/master)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Review by babayK0 — CHANGES_REQUESTED** [review]
Good job :100: 

Lets do some improvements)

First of all, your code contains a lot of nested array operations. What should you do if count of elements will be hundreds of thousands ?
 
See my comments below and fix this one please`GET https://ustymchyk.github.io/favicon.ico 404`

**Comment by ustymchyk** [comment]
@babayK0 

I rethought the implementation of this task.
Instead of creating elements, I`m generating DOMString.
Now code is much more readable and has less code.
Could you review it one more time, please?

File: submissions/ustymchyk/js-dom/index.js [hunk 1]
```diff
@@ -10,100 +10,69 @@ function init() {
...
+          ${type}
+        </button>
+      </li>
+    `;
+  }).join('');
+}
 
-    return listItem;
+function addListenersToMenu() {
+  document.querySelectorAll('.menu__item').forEach(btn => {
```
Comment (babayK0) [inline submissions/ustymchyk/js-dom/index.js:64]: Maybe one event listener for wrapper will be better.

**Review by babayK0 — CHANGES_REQUESTED** [review]
Wow, good rework !

But lets do one more step. 
What about changing your function declaration to function expressions ? With them your code will be more clean and readable)

**Comment by ustymchyk** [comment]
done
one more review, please) @babayK0

**Review by babayK0 — APPROVED** [review]
Good job :+1: 

P.S. My personal like for pokenons :smile_cat:

**Pull request merged**